core/templates/dev/head/pages/story-editor-page/editor-tab/story-node-editor.directive.ts

```diff
@@ -76,6 +79,20 @@ angular.module('oppia').directive('storyNodeEditor', [
               }
             }
           };
+          var _isExpIdValid = function(expId) {
```

Comment:
Since this whole function is basically the same as the one in story-editor-navbar, except that, that checks for all explorations, can you add these checks to the `validate` function in StoryNodeObjectFactory and just use that here? Once that is done, you could remove the `_validateExplorations` function in story-editor-navbar as well.

Reply:
Hi @aks681, The StoryNodeObjectFactory is written in Angular and the `ExplorationSummaryBackendApiService` that validates the expId is written in AngularJs. So I'm not sure if we can do that.

Reply:
Hmm, not sure if we need to create a service for just the one check. Actually, I added another check in #8581, in the other function which cannot be reused here. I think it's fine to leave this as is then. 
@DubeySandeep What do you think?

Reply:
It is fine to make a service for just the one check. It will have one check now but it may grow to have other checks later, depending on what its scope is.

One suggestion I have though is to add Async to the end of the name, so that it's clear that this is an asynchronous method.

Reply:
@seanlip Ok, So what should the service be called?

/cc @aks681 @DubeySandeep

Reply:
Finding good names for things is an important part of software development. Try and propose something, and I can give you feedback. Make the name clearly convey what the service is, and try to follow existing conventions in the codebase.

Reply:
@seanlip Added another service with name `exploration-id-validation.service.ts`.

Reply:
That works. Thanks!